1_imagestream_data_gen/imagestream_data_gen.py

```diff
@@ -0,0 +1,140 @@
+            "cluster detected.")
+        print("[!] [WARNING] Detected k8s version: {}\n".format(
+            dyn_client.version.get("kubernetes", {}).get("gitVersion", "")))
+        selection = input("[?] Press 'Enter' to quit, or type 'i' to ignore warning: ")
+        if 'i' not in selection:
+            print("Exiting...")
+            exit(1)
+except Exception as e:
+    print("[!] Failed to parse OpenShift version.")
+    print(e)
+
+
+# Object serving as 'get' default for empty results
+class EmptyK8sResult:
+    __dict__ = {}
+
+
+emptyDict = EmptyK8sResult()
+
+# Make output dir if doesn't exist
+if not os.path.exists(output_dir):
+    os.makedirs(output_dir)
+
+with open(script_dir + '/vars/imagestream-data-gen.yml') as f:
+    input_data = yaml.load(f, Loader=yaml.FullLoader)
+
+node_list = []
+output = []
+
+for namespace in input_data['namespace_to_migrate']:
+    print("Processing namespace: [{}]".format(namespace))
+    v1_namespaces = dyn_client.resources.get(api_version='v1', kind='Namespace')
+    try:
+        ns = v1_namespaces.get(name=namespace)
+        ns_out = {'namespace': namespace, 'annotations': ns.metadata.get("annotations", emptyDict).__dict__}
+        output.append(ns_out)
+    except Exception as e:
+        # TODO: figure out a way to report the missing namespace
+        print("\n[WARNING] v1/namespace not found: {}\n".format(namespace))
+        print(e)
+
+ns_data_file = os.path.join(output_dir, 'namespace-data.json')
+with open(ns_data_file, 'w') as f:
+    json.dump(output, f, indent=4)
+    print("[!] Wrote {}".format(ns_data_file))
+
+namespaces = output
+output = []
+
+for item in namespaces:
+    namespace = item["namespace"]
+    print("Processing ImageStreams for namespace: [{}]".format(namespace))
+
+    imagestreams = dyn_client.resources.get(api_version='image.openshift.io/v1', kind='ImageStream')
+    imagestreams_list = imagestreams.get(namespace=namespace)
+
+    for imagestream in imagestreams_list.items:
+        # print(imagestream)
+        tags_to_process = []
+        tags_to_migrate = []
+        if imagestream.status.tags is None:
+            continue
+        if imagestream.spec.tags is None:
+            for tag in imagestream.status.tags:
+                tags_to_process.append(tag.name)
+        else:
+            for tag in imagestream.spec.tags:
+                if tag.kind == "ImageStreamTag":
+                    continue
+                tags_to_process.append(tag.name)
+        # print(imagestream.metadata.name, tags_to_process)
+        source_registry_url = imagestream.status.dockerImageRepository.split("/")[0]
+        for tag in imagestream.status.tags:
+            for image in tag.items:
+                # print(image.dockerImageReference)
+                image_reference_split = image.dockerImageReference.split("/")
+                if len(image_reference_split) < 3 or source_registry_url == image_reference_split[0]:
```

Comment:
I'm not quite sure I understand the scenario where image_reference_split has fewer than 3 elements. In this case we process the tag and set source_registry_in_image_tag to false. Although if something happened and there were more than 3 elements, then we'd end up setting source_registry_in_image_tag to true (probably incorrectly). I'm not sure if it's possible for this to occur, though.

Reply:
fixed this in #10